composer/algorithms/progressive_resizing/progressive_resizing.py

```diff
@@ -44,42 +46,61 @@ def resize_batch(X: torch.Tensor,
             ``resize_targets`` is ``True``,
         scale_factor: scaling coefficient for the height and width of the
             input/output tensor. 1.0 keeps the original size.
         mode: type of scaling to perform. Value must be one of ``'crop'`` or
             ``'resize'``. ``'crop'`` performs a random crop, whereas ``'resize'``
-            performs a bilinear interpolation.
+            performs a nearest neighbor interpolation.
         resize_targets: whether to resize the targets, ``y``, as well
 
     Returns:
         X_sized: resized input tensor of shape ``(N, C, H * scale_factor, W * scale_factor)``.
         y_sized: if ``resized_targets`` is ``True``, resized output tensor
-            of shape ``(N, C, H * scale_factor, W * scale_factor)``. Otherwise
-            returns original ``y``.
+            of shape ``(N, H * scale_factor, W * scale_factor)`` or  ``(N, C, H * scale_factor, W * scale_factor)``.
+            Depending on the input ``y``. Otherwise returns original ``y``.
     """
     # Short-circuit if nothing should be done
     if scale_factor >= 1:
         return X, y
 
-    def resize_tensor(tensor: torch.Tensor) -> torch.Tensor:
-        # Reduce the size of input images, either via cropping or downsampling
-        if mode.lower() == "crop":
-            Hc = int(scale_factor * tensor.shape[2])
-            Wc = int(scale_factor * tensor.shape[3])
-            resize_transform = transforms.RandomCrop((Hc, Wc))
-        elif mode.lower() == "resize":
-            resize_transform = partial(F.interpolate, scale_factor=scale_factor, mode='bilinear')
-        else:
-            raise ValueError(f"Progressive mode '{mode}' not supported.")
-        return resize_transform(tensor)
-
-    X_sized = resize_tensor(X)
-    if resize_targets:
-        y_sized = resize_tensor(y)
+    # Prep targets for resizing if necessary
+    if check_for_index_targets(y) and resize_targets is True:
+        # Add a dimension to match shape of the input and change type for resizing
+        y_sized = y.float().unsqueeze(1)
     else:
         y_sized = y
+
+    if mode.lower() == "crop" and resize_targets is False:
+        # Make a crop transform for X
+        resize_transform = _make_crop(tensor=X, scale_factor=scale_factor)
+        X_sized, y_sized = resize_transform(X), y
+    elif mode.lower() == "crop" and resize_targets is True:
+        # Make a crop transform for X and y
+        resize_transform, resize_y = _make_crop_pair(X=X, y=y_sized, scale_factor=scale_factor)
+        X_sized, y_sized = resize_transform(X), resize_y(y_sized)
+    elif mode.lower() == "resize":
+        # Make a resize transform (can be used for X or y)
+        resize_transform = _make_resize(scale_factor=scale_factor)
+        X_sized = resize_transform(X)
+        if resize_targets:
+            y_sized = resize_transform(y_sized)
+    else:
+        raise ValueError(f"Progressive mode '{mode}' not supported.")
+        X_sized, y_sized = X, y
```

Comment:
```suggestion
```
This won't be reached since you raised an error

Reply:
Fixed